.github/workflows/delploy_cdk_stack.yml

```diff
@@ -0,0 +1,86 @@
+name: Deploy cdk stack
+
+permissions:
+  id-token: write
+  contents: read
+
+on:
+  workflow_call:
+    inputs:
+      stage:
+        required: true
+        type: string
+    secrets:
+      target-account:
+        required: true
+
+jobs:
+  deploy-cdk-stack:
+    runs-on: ubuntu-latest
+    needs:
+      - build-layer
```

Comment:
left over? I believe this will fail because there isn't a job `build-layer` defined within this workflow to depend on

Reply:
yes, it should be outside when we call the step.